cea/technologies/thermal_network/thermal_network.py

```diff
@@ -2333,13 +2354,54 @@ def solve_network_temperatures(thermal_network, t):
+                    t_supply_nodes_2__k, plant_node, \
+                    q_loss_edges_2_supply_kW, _ = calc_supply_temperatures(t, edge_node_df.copy(),
+                                                                           edge_mass_flow_df_2_kgs, k, thermal_network)
+                    # check if all substation temperatures are satisfied
+                    dt_nodes = t_supply_nodes_2__k - 273.15 - thermal_network.t_target_supply_df.loc[t]
+                    dt_nodes_max = dt_nodes.max().copy()
+                    dt_tolerance = 0.00001 # TODO: defined by users
+                    # identify the nodes
+                    nodes_insufficient = dt_nodes[dt_nodes > dt_tolerance].index
+                    if dt_nodes_max >= dt_tolerance and VF_iter < 10:
+                        print('maximum dT at substations: ',dt_nodes_max) # FIXME: to be removed (check if it's reducing)
+                        # increase node flows
+                        substations_nodes_df_old = mass_flow_substations_nodes_df.copy()
+                        for node in nodes_insufficient:
+                            mass_flow_substations_nodes_df[node] = substations_nodes_df_old[node] * 1.1 # increase flow by 10%
+                        # solve for the required mass flow rate on each edge/pipe
+                        edge_mass_flow_df_2_kgs = calc_mass_flow_edges(edge_node_df.copy(),
+                                                                       mass_flow_substations_nodes_df,
+                                                                       thermal_network.all_nodes_df,
+                                                                       thermal_network.pipe_properties[:][
+                                                                       'D_int_m':'D_int_m'].values[0],
+                                                                       thermal_network.edge_df['pipe length'], t_edge__k)
+                        VF_iter = VF_iter + 1
+                    elif dt_nodes_max >= dt_tolerance and VF_iter >= 10:
+                        for node in nodes_insufficient:
+                            index_insufficient = np.argwhere(edge_node_df.index == node)[0][0]
+                            t_target_supply__c = thermal_network.t_target_supply_df.loc[t]
+                            t_supply_nodes_2__k[index_insufficient] = t_target_supply__c[index_insufficient] + 273.15
+                            # force setting node temperature to target to avoid substation HEX calculation error.
+                            # However, it might potentially cause error at mass flow iteration.
+                            print('force node: ', node, 'with dt=', dt_nodes[node], ' at time: ', t)
+                        VF_flag = False
+                    else:
+                        VF_flag = False
+
+            else:
+                raise ('control strategy not specified: ', control)
```

Comment:
this is a very old school style of exception. i'm not even sure if that really works. just tested it in ipython:
```
In [2]: control = 'test'

In [3]: raise ('control strategy not specified: ', control)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-0cc27d99a90f> in <module>()
----> 1 raise ('control strategy not specified: ', control)

TypeError: exceptions must be old-style classes or derived from BaseException, not str
```

so this is a bug. you could use  `raise ValueError('control strategy not specified: {control}'.format(control=control))`

Reply:
Actually, `control` is not defined in the code, either, so this is a double bug. Fixed that too.

Reply:
Actually, this line doesn't even matter, because the temperature control never gets read as any other option (see comment in the bottom).